### DISCUSSION TITLE
the visualization of breast density challenge tutorial using tensorboard

### DISCUSSION THREAD

**Discussion by GhanemBahrini01** [Project-MONAI/tutorials#854] [Q&A] [answered]
Hello all,

So I have run the tutorial code https://github.com/Project-MONAI/tutorials/tree/main/federated_learning/breast_density_challenge and when visualizing the results I was wondering what is the difference between  " val_local_model_acc " and " val_acc_local_model_acc " (the visualization result after running " tensorboard --logdir=./result_server")!

**Comment by holgerroth** [comment, marked as answer]
Thank you for your question. `val_local_model_acc`  and  `val_acc_local_model_acc` are computed at different phases of training. Metrics with `val_local_model_*` are computed after the local [training loop](https://github.com/Project-MONAI/tutorials/blob/0da3ee59cdeebdafa5ef262d5fbc8745b364e3dd/federated_learning/breast_density_challenge/code/pt/learners/mammo_learner.py#L409) and metrics with `val_acc_local_model_*` are also computed during the local training at certain intervals (controlled by val_freq). 
If [aggregation_epochs](https://github.com/Project-MONAI/tutorials/blob/0da3ee59cdeebdafa5ef262d5fbc8745b364e3dd/federated_learning/breast_density_challenge/code/configs/mammo_fedavg/config/config_fed_client.json#L34) = 1, then both plots will be the same. 

Note, `val_global_model_kappa` is calculated separately with each site's data at the start of each round, which is why you see 3 separate traces. If you look at the [leaderboard](https://docs.google.com/spreadsheets/d/1IN7woElI55XmzNrAsw5yX-fZxRyfItIPei0w6nt-yfQ/edit#gid=0), you will see that statistics are tracked both separately for each site (i.e. Lin. Kappa 1-3) and averaged across all sites (i.e. Lin. Kappa all).
